### PR TITLE
Appnexus targeting function

### PR DESCRIPTION
**Pull request by jaiminpanchal27**
## Type of change

- [ ] Bugfix
- [x ] Feature
- [ ] New bidder adapter
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other

## Description of change

Targeting function to add keywords for appnexus added.

### CODE REVIEW

**Comment by mkendall07** [comment]
Looks like tests are failing.

**Comment by jaiminpanchal27** [comment]
@mkendall07 you can take a look now.

File: src/targeting.js [hunk 1]
```diff
@@ -0,0 +1,181 @@
+import { uniques, isGptPubadsDefined, getHighestCpm, adUnitsFilter } from './utils';
+var bidmanager = require('./bidmanager.js');
```
Comment (mkendall07) [inline src/targeting.js]: prefer const.

File: src/prebid.js [hunk 2]
```diff
@@ -403,9 +243,19 @@ $$PREBID_GLOBAL$$.setTargetingForGPTAsync = function () {
...
 
   //first reset any old targeting
-  resetPresetTargeting();
+  targeting.resetPresetTargeting();
   //now set new targeting keys
-  setTargeting(getAllTargeting());
+  targeting.setTargeting(targeting.getAllTargeting());
+};
+
+$$PREBID_GLOBAL$$.setTargetingForAn = function() {
```
Comment (mkendall07) [inline src/prebid.js]: Let's rename to `setTargetingForAst`

File: src/utils.js [hunk 3]
```diff
@@ -542,6 +542,12 @@ export function isGptPubadsDefined() {
   }
 }
 
+export function isApntagDefined() {
```
Comment (mkendall07) [inline src/utils.js]: probably makes sense to move this into `targeting.js`

File: src/targeting.js [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+
+targeting.setTargetingForAn = function() {
+  let targeting = $$PREBID_GLOBAL$$.getAdserverTargeting();
+  Object.keys(targeting).forEach(targetId =>
+    Object.keys(targeting[targetId]).forEach(key => {
+      utils.logMessage(`Attempting to set targeting for targetId: ${targetId} key: ${key} value: ${targeting[targetId][key]}`);
+      //setKeywords supports string and array as value
+      if(utils.isStr(targeting[targetId][key]) || utils.isArray(targeting[targetId][key])) {
+        var keywordsObj = {};
+        keywordsObj[key] = targeting[targetId][key];
```
Comment (mkendall07) [inline src/targeting.js]: need to check for the `hb_adid` key and convert to `HB_ADID`

Comment (mkendall07) [inline src/targeting.js]: We also need to multiple the value of `hb_pb` by 100 and trim it (no decimals). `(value *  100).toFixed();`

**Comment by mkendall07** [comment]
LGTM.

**Pull request merged**